src/phockup.py

```diff
@@ -53,16 +55,45 @@ def check_directories(self):
 
     def walk_directory(self):
         """
-        Walk input directory recursively and call process_file for each file except the ignored ones
+        Walk input directory recursively, split the file list into parts and process them in threads
         """
+        files_temp = []
         for root, _, files in os.walk(self.input):
-            files.sort()
             for filename in files:
                 if filename in ignored_files:
                     continue
 
                 file = os.path.join(root, filename)
-                self.process_file(file)
+                files_temp.append(file)
+
+        files = files_temp
+        files.sort()
+        num_files = len(files)
+        num_threads = min(self.threads, num_files)
+
+        if num_threads > 1:
+            threads = []
+            for i in range(0, num_threads):
+                files_part = files[i::num_threads]
+                t = threading.Thread(target=self.process_file_worker, args=(files_part,))
```

Comment:
Please avoid using single character variable names. `trhead` will work fine here

Reply:
Ok, that's true.